common/djangoapps/student/models.py

```diff
@@ -823,10 +823,17 @@ def is_course_full(self, course):
 
     def users_enrolled_in(self, course_id):
         """Return a queryset of User for every user enrolled in the course."""
-        return User.objects.filter(
+        course_users = User.objects.filter(
```

Comment:
Did you consider using select_related to UserProfile here and discount that approach for some reason?